sort.go

```diff
@@ -0,0 +1,202 @@
+package goset
+
+import (
+	"reflect"
+	"sort"
+
+	"golang.org/x/exp/maps"
+)
+
+// sortSet is adapted from (the inaccessible because internal) fmtsort.Sort.
+// It returns a []T of the values in a Set[T], in a stable sorted order according to the following rules.
+//
+// TODO(daynemay): look into the "issues raised by unorderable key values such as NaNs" (as mentioned in fmtsort.Sort)
+//
+// The ordering rules are more general than with Go's < operator:
+//
+//  - when applicable, nil compares low
+//  - ints, floats, and strings order by <
+//  - NaN compares less than non-NaN floats
+//  - bool compares false before true
+//  - complex compares real, then imag
+//  - pointers compare by machine address
+//  - channel values compare by machine address
+//  - structs compare each field in turn
+//  - arrays compare each element in turn
+//    Otherwise identical arrays compare by length.
+//  - interface values compare first by reflect.Type describing the concrete type
+//    and then by concrete value as described in the previous rules.
+//
+func sortSet[T comparable](theSet Set[T]) []T {
```

Comment:
I'd like to change the signature of this function so that it doesn't contain any reference to `Set` type.

I could see it potentially having value as its own package; keeping `Set` out of here would make that change easier if we decide to go in that direction.